source/routing/raptor.cpp

```diff
@@ -673,16 +749,34 @@ void RAPTOR::raptor_loop(Visitor visitor, const nt::RTLevel rt_level, uint32_t m
          * We want to do it, to favoritize normal vj against stay_in vjs
          */
         for (auto q_elt : Q) {
+            /// We will scan the journey_pattern q_elt.first, starting from its stop numbered q_elt.second
+
             const JpIdx jp_idx = q_elt.first;
+
+            const RouteIdx route_idx = data.dataRaptor->jp_container.get(jp_idx).route_idx;
+
+            /// q_elt.second == visitor.init_queue_item() means that
+            /// this journey_pattern is marked "not to be scanned"
             if (q_elt.second != visitor.init_queue_item()) {
+                /// we begin scanning the journey_pattern as if we were not yet aboard a vehicle
                 bool is_onboard = false;
                 DateTime workingDt = visitor.worst_datetime();
                 DateTime base_dt = workingDt;
-                typename Visitor::stop_time_iterator it_st;
+                DateTime working_walking_duration = DateTimeUtils::not_valid;
+                SpIdx boarding_stop_point = SpIdx();
+
+                /// will be used to iterate through the StopTimeS of
+                /// the vehicle journey of the current journey_pattern (jp_idx)
+                ///  with the relevant departure date
+                typename Visitor::stop_time_iterator it_st;  /// item = type::StopTime
                 uint16_t l_zone = std::numeric_limits<uint16_t>::max();
+
+                LOG4CPLUS_TRACE(logger, " Scanning line  " << data.pt_data->routes[route_idx.val]->line->uri);
+
                 const auto& jpps_to_explore =
                     visitor.jpps_from_order(data.dataRaptor->jpps_from_jp, jp_idx, q_elt.second);
                 for (const auto& jpp : jpps_to_explore) {
+                    /// jpp : dataRAPTOR::JppsFromJp::Jpp
```

Comment:
If this comment does help you, you can replace `auto` with the actual type. Comment doesn't usually age well :older_man:  ^^

Reply:
Good idea ! Done https://github.com/CanalTP/navitia/pull/2937/commits/53cd072b2a5f6596daa177532ad871f67da15f12